src/serlio/prtMaterial/MaterialInfo.cpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * Serlio - Esri CityEngine Plugin for Autodesk Maya
+ *
+ * See https://github.com/esri/serlio for build and usage instructions.
+ *
+ * Copyright (c) 2012-2019 Esri R&D Center Zurich
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "prtMaterial/MaterialInfo.h"
+
+#include <cmath>
+
+double MaterialColor::r() const noexcept {
+	return (*this)[0];
+}
+
+double MaterialColor::g() const noexcept {
+	return (*this)[1];
+}
+
+double MaterialColor::b() const noexcept {
+	return (*this)[2];
+}
+
+double MaterialTrafo::su() const noexcept {
+	return (*this)[0];
+}
+
+double MaterialTrafo::sv() const noexcept {
+	return (*this)[1];
+}
+
+double MaterialTrafo::tu() const noexcept {
+	return (*this)[2];
+}
+
+double MaterialTrafo::tv() const noexcept {
+	return (*this)[3];
+}
+
+double MaterialTrafo::rw() const noexcept {
+	return (*this)[4];
+}
+
+std::array<double, 2> MaterialTrafo::tuv() const noexcept {
+	return {tu(), tv()};
+}
+
+std::array<double, 3> MaterialTrafo::suvw() const noexcept {
+	return {su(), sv(), rw()};
+}
+
+MaterialColor MaterialInfo::getColor(adsk::Data::Handle sHandle, const std::string& name) {

```

Comment:
consider passing const ref for sHandle, it's only read
(same below)

Reply:
Unfortunately, `const` is not possible, since we need to call the non-const member function `setPositionByMemberName` on the handle. I made them refs inside `MaterialInfo`.